orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py

```diff
@@ -5,13 +5,23 @@
 
 import sys
 import torch
+import torch.utils.checkpoint
 import warnings
 from torch.onnx import symbolic_helper
 
 from onnxruntime.capi._pybind_state import register_torch_autograd_function
 from ._fallback import _FallbackManager, ORTModuleONNXModelException, ORTModuleTorchModelException, wrap_exception
 from . import _logger
 
+# Some autograd.Function's shouldn't be exported as PythonOp.
+# If CheckpointFunction is exported as PythonOp, the checkpointed computation
+# may be computed by Pytorch, not ORT. This situation is especially important
+# for big models such as GPT-2. Exporting CheckpointFunction as PythonOp means
+# every transformer would be computed by Pytorch and ORT doesn't contribute
+# at all.
+_banned_autograd_function_names = set([torch.utils.checkpoint.CheckpointFunction.__name__])
```

Comment:
I am curious, is that possible we detect other checkpoint activation implementations for example deepspeed?

Reply:
```suggestion
BLACKLIST_AUTOGRAD_FUNCTION_NAMES = set([torch.utils.checkpoint.CheckpointFunction.__name__])
```
Constants names are usually all caps by convention

Reply:
Color is not good in name. I will still use "BANNED".